bundles/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/bus/KNXBinding.java

```diff
@@ -107,9 +110,21 @@ public void removeKNXTypeMapper(KNXTypeMapper typeMapper) {
      */
     @Override
     protected void internalReceiveCommand(String itemName, Command command) {
-        logger.trace("Received command (item='{}', command='{}')", itemName, command.toString());
-        if (!isEcho(itemName, command)) {
-            writeToKNX(itemName, command);
+        logger.trace(
+                "1. internalReceiveCommand from openHAB (item='{}', command='{}', operatingEnvironmentIsFelix = '{}'",
+                itemName, command.toString(), operatingEnvironmentIsFelix);
+
+        // openHAB1: After receiving a command from openHAB1, NO update event is triggered later.
+        // openHAB2: After receiving a command from openHAB2, ALWAYS update event is triggered later.
+        // In openHAB2 we ignore this command and wait for receiving the update event from openHAB,
+        // then only this is sent to the KNX bus by openHAB2
+
+        if (getEnvironmentIsFelix() == true) {
+            // ignore internalReceiveCommand in openHAB 2
```

Comment:
I share your worries - I am still of the opinion that this needs more thorough analysis; unfortunately, I didn't find time to look into it any further myself yet.